app.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+)
+
+type Flags struct {
+	countStrings   bool
+	showDuplicates bool
+	haveRegister   bool
+	showUnique     bool
+	countSymbols   int
+	countRow       int
+}
+
+type Data struct {
+	command    *Flags
+	inputFile  io.Reader
+	outputFile io.Writer
+}
+
+func Uniq(data *Data) error {
+	var err error = nil
+	var prevValue string
+	originPrevValue := ""
+
+	lenghtStrings := 0
+	lastCount := 0
+
+	haveRows := (*data.command).countRow
```

Comment:
В golang не обязательно разыменовывать указатели, как в C/C++ - можно через точку обращаться к полям
```suggestion
	haveRows := (data.command).countRow
```

В остальных местах аналогично, плиз